src/fixture-factory.ts

```diff
@@ -0,0 +1,114 @@
+import {
+    FactoryBuildOptions,
+    FactoryFunction,
+    FactoryOptions,
+    FixtureStatic,
+} from './types';
+import { TypeFactory } from './type-factory';
+import {
+    fileAppendJson,
+    readFileIfExists,
+    saveFixture,
+    structuralMatch,
+} from './utils';
+
+type WriteBuildAsync<T> = (
+    path: string,
+    options?: FactoryBuildOptions<T>,
+) => Promise<T>;
+type WriteBuildSync<T> = (path: string, options?: FactoryBuildOptions<T>) => T;
+type WriteBatchAsync<T> = (
+    path: string,
+    size: number,
+    options?: FactoryBuildOptions<T>,
+) => Promise<T[]>;
+type WriteBatchSync<T> = (
+    path: string,
+    size: number,
+    options?: FactoryBuildOptions<T>,
+) => T[];
+
+type PropsBuildAsync = { batch: false; synchronous: false };
+type PropsBuildSync = { batch: false; synchronous: true };
+type PropsBatchAsync = { batch: true; synchronous: false };
+type PropsBatchSync = { batch: true; synchronous: true };
+
+const propsBuildAsync = { batch: false, synchronous: false };
+const propsBuildSync = { batch: false, synchronous: true };
+const propsBatchAsync = { batch: true, synchronous: false };
+const propsBatchSync = { batch: true, synchronous: true };
+
+export class FixtureFactory<T> extends TypeFactory<T> {
+    private defaultPath = '';
+
+    constructor(
+        defaults: FactoryOptions<T>,
+        factory?: FactoryFunction<T>,
+        defaultPath?: string,
```

Comment:
Should be required param:

```suggestion
    constructor(   
        defaultPath: string,
        defaults: FactoryOptions<T>,
        factory?: FactoryFunction<T>,
```